File: server-mh/src/Match/Handler/Models/Piece.ts
import { Cell } from "./Cell";
import { PieceState } from "./PieceState";
import{Player} from "./Player";
export class Piece {
    public readonly id: number;
    public readonly player: Player;
    public currentCell: Cell;
    public readonly initialCell: Cell;
    public pieceState: PieceState;


    constructor(
        id: number,
        initialCell: Cell,
        player: Player
    ) {
        this.id = id;

        this.initialCell = initialCell;
        this.currentCell = initialCell;
        this.pieceState = new PieceState();

        this.player = player;
    }

   
}

File: Client-unity/Ludo/Assets/Scripts/OfflineMatchHandler/Piece.cs
namespace Ludo.Offline
{
    public class Piece
    {
        public readonly int Id;
        public readonly Player Player;

        public Cell CurrentCell;
        public readonly Cell InitialCell;

        public PieceState PieceState;

        public Piece(
            int id,
            Cell initialCell,
            Player player)
        {
            Id = id;
            InitialCell = initialCell;
            CurrentCell = initialCell;
            Player = player;

            PieceState = new PieceState();
        }
    }
}